### PR TITLE
Update template to use a store

### PR DESCRIPTION
**Pull request by dramich**
Problem:
Templates need their versionsLinks updated based off the rancher server
for both list and watch calls

Solution:
Add a Transformer to add the links and move the logic out of a
formatter

https://github.com/rancher/rancher/issues/22835

### CODE REVIEW

File: pkg/api/customization/catalog/store.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	ts.Store = s
+
+	return ts
+}
+
+func (t *templateStore) extractVersionLinks(apiContext *types.APIContext, resource map[string]interface{}) map[string]interface{} {
+	schema := apiContext.Schemas.Schema(&managementschema.Version, client.TemplateVersionType)
+	r := map[string]interface{}{}
+	versionMap, ok := resource[client.CatalogTemplateFieldVersions].([]interface{})
+	if ok {
```
Comment (daxmc99) [inline pkg/api/customization/catalog/store.go]: Is it ok if we return an empty map here?

**Comment by cbron** [comment]
Commits should probably be squashed

File: tests/integration/suite/test_catalog.py [hunk 2]
```diff
@@ -177,13 +177,6 @@ def test_template_version_links(admin_mc, admin_pc, custom_catalog,
 
     assert len(templates.data[0]['versionLinks']) == 0
 
-    # If no rancher version is set get back both versions
-    templates = client.list_template(catalogId=c_name)
-
-    assert len(templates.data[0]['versionLinks']) == 2
-    assert '1.6.0' in templates.data[0]['versionLinks']
-    assert '1.6.2' in templates.data[0]['versionLinks']
-
```
Comment (cbron) [inline tests/integration/suite/test_catalog.py]: We could create a test here with an out-of-scope version right ?

File: pkg/api/customization/catalog/store.go (hunk 1 above)
Comment (dramich) [reply pkg/api/customization/catalog/store.go]: Yes, that's the same as having no versions matching your current rancher version.

File: tests/integration/suite/test_catalog.py (hunk 2 above)
Comment (dramich) [reply tests/integration/suite/test_catalog.py]: There is above this.

**Comment by daxmc99** [comment]
LGTM

**Pull request merged**